### ISSUE TITLE
Wrong value of custom input template variable in CloudWatch

### ISSUE THREAD

**Issue by thwblueboy** [grafana/grafana#11050]
- What Grafana version are you using?  Grafana v4.6.3 (commit: 7a06a47) 
- What datasource are you using? Cloudwatch
- What OS are you running grafana on? macOS
- What did you do? 
I have added a template variable with "custom" type and use it in Cloudwatch and text panel. When I select one in this template variable, it works will in both Cloudwatch and text panel:
![image](https://user-images.githubusercontent.com/2154478/36839199-4e079c44-1d7c-11e8-8c50-32cc08c59973.png)

But when I input some custom value into this template variable, Cloudwatch is still using its previous value but text panel uses the new one:
![image](https://user-images.githubusercontent.com/2154478/36839311-b806e3f2-1d7c-11e8-9ece-bfa19249ca9e.png)

- What was the expected result? The Cloudwatch datasource should use the new value as well (custom input).

**Comment by davidnewhall** [comment]
I'm seeing this happen in Grafana 6.3.3 using only custom variables (no data source). When I change the value (read: a single value in my test), save the dashboard and reload the page: the old value is still shown in the UI. 

This is in addition to custom values not performing variable interpolation. If I can't figure that out I'll open another issue.

**Comment by sunker** [comment]
I can't reproduce this. Tested in master and in 5.4. What I did was: 
1. Create a custom variable
2. Use the custom variable in a query editor --> works fine
3. Add a new value to the custom variable
4. The new value is being interpreted correctly. Query returns no data when inputing a bad value to the custom template variable, and returns query data when using a valid input value. 

Also, after adding a new value and hit Save dashboard, the new value is still selected when reloading the dashboard. 

Are you still experiencing this @davidnewhall, @thwblueboy and @farthinder? Am i testing this in a correct way?
